code/datums/revision.dm

```diff
@@ -11,30 +11,77 @@ GLOBAL_PROTECT(revision_info) // Dont mess with this
 	var/commit_hash
 	/// Date that this commit was made
 	var/commit_date
+	/// Origin commit (Only set if running TGS, and will only be different if the server is running testmerges)
+	var/origin_commit
+	/// List of testmerges (If applicable)
+	var/list/testmerges = list()
 
+// Pull info from the rust DLL
 /datum/code_revision/New()
 	commit_hash = rustg_git_revparse("HEAD")
 	if(commit_hash)
 		commit_date = rustg_git_commit_date(commit_hash)
 
+// Pull info from TGS
+/datum/code_revision/proc/load_tgs_info()
+	testmerges = world.TgsTestMerges()
+	var/datum/tgs_revision_information/revinfo = world.TgsRevision()
+	if(revinfo)
+		commit_hash = revinfo.commit
+		origin_commit = revinfo.origin_commit
+		commit_date = rustg_git_commit_date(commit_hash)
+
 /**
   * Code Revision Logging Helper
   *
   * Small proc to simplify logging all this stuff
   */
 /datum/code_revision/proc/log_info()
 	// Put revision info in the world log
-	var/logmsg
+	var/list/logmsgs = list()
 	if(commit_hash && commit_date)
-		logmsg = "Running ParaCode commit: [commit_hash] (Date: [commit_date])"
+		logmsgs += "Running ParaCode commit: [commit_hash] (Date: [commit_date])"
 	else
-		logmsg = "Unable to determine revision info! Code may not be running in a git repository."
+		logmsgs += "Unable to determine revision info! Code may not be running in a git repository."
+
+	// Check if we were on TGS
+	if(world.TgsAvailable())
+		if(origin_commit && (commit_hash != origin_commit)) // Commits are different, theres likely a testmerge
+			logmsgs += "Origin commit: [origin_commit]"
+		if(length(testmerges))
+			logmsgs += "The following PRs are testmerged:"
+			for(var/pr in testmerges)
+				var/datum/tgs_revision_information/test_merge/tm = pr
+				logmsgs += "PR #[tm.number] at commit [tm.pull_request_commit]"
+				// Log these in blackbox so they can be attributed to round IDs easier in the future
+				SSblackbox.record_feedback("associative", "testmerged_prs", 1, list("number" = "[tm.number]", "commit" = "[tm.pull_request_commit]", "title" = "[tm.title]", "author" = "[tm.author]"))
+
+	var/logmsg = logmsgs.Join("\n")
 
 	// Log it in all these
 	log_world(logmsg)
 	log_runtime_txt(logmsg)
 	log_runtime_summary(logmsg)
 
+
+/**
+  * Testmerge Chat Message Helper
+  *
+  * Formats testmerged PRs into a nice message
+  * Arguments:
+  * * header - Should a header be sent too
+  */
+/datum/code_revision/proc/get_testmerge_chatmessage(header = FALSE)
+	var/list/msg = list()
+	if(header)
+		msg += "<span class='notice'>The following PRs are currently testmerged:</span>"
+
+	for(var/pr in GLOB.revision_info.testmerges)
+		var/datum/tgs_revision_information/test_merge/tm = pr
+		msg += "- PR <a href='[config.githuburl]/pull/[tm.number]'>#[tm.number] - [tm.title]</a>"
```

Comment:
```suggestion
		msg += "- PR <a href='[tm.url]'>#[tm.number] - [tm.title]</a>"
```

Reply:
Will do when I get around to sorting this again